### PR TITLE
Feature/bau 341 add new versioning columns to release table

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
This PR needs to go in after the PR for EES-322 - https://github.com/dfe-analytical-services/explore-education-statistics/pull/1385.

This PR:
- adds versioning fields to Release and associated columns to the database

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/Versioned.cs [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-        T Current { get; }
+        public DateTime Created { get; set; }
+        
+        public User CreatedBy { get; set; } 
+        
+        public Guid CreatedById { get; set; }
+        
+        public TSelf Original { get; set; }
 
+        public Guid OriginalId { get; set; }
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model/Versioned.cs]: Should this be optional?
I can see the seeded Releases will be getting an Id of new Guid("00000000-0000-0000-0000-000000000000").

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/ContentDbContextModelSnapshot.cs [hunk 2]
```diff
@@ -1447,51 +1456,70 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
                             TimePeriodCoverage = "AY",
-                            TypeId = new Guid("9d333457-9132-4e55-ae78-c55cb3673d7c")
+                            TypeId = new Guid("9d333457-9132-4e55-ae78-c55cb3673d7c"),
+                            Version = 0
                         },
                         new
                         {
                             Id = new Guid("63227211-7cb3-408c-b5c2-40d3d7cb2717"),
+                            Created = new DateTime(1, 1, 1, 0, 0, 0, 0, DateTimeKind.Unspecified),
+                            CreatedById = new Guid("00000000-0000-0000-0000-000000000000"),
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/ContentDbContextModelSnapshot.cs]: Is there any superuser id that can be used in the ContentDbContext seed Release data for the CreatedById field?

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/Versioned.cs (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Model/Versioned.cs]: Good spot, but I will set the correct value for the Releases in ContentDbContext and make sure that that makes it through into the Snapshot file too.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/ContentDbContextModelSnapshot.cs (hunk 2 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/ContentDbContextModelSnapshot.cs]: Yup will set appropriate values in ContentDbContext, thanks!

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/Versioned.cs (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Model/Versioned.cs]: Added new migration to updated these

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/ContentDbContextModelSnapshot.cs (hunk 2 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/ContentDbContextModelSnapshot.cs]: Added new migration to update these

**Pull request merged**